modules/common/react/lib/utils/mergeCallback.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Merge callbacks by returning a function that calls both functions. This is useful for components
+ * that need to provide callback functionality, but want to allow additional functionality to be
+ * provided
+ * @example
+ * const MyComponent = ({ myProp, ...elemProps }) => {
+ *   return (
+ *     <button onClick={mergeCallback(cb, elemProps.onClick)}>Click Me</button>
+ *   )
+ * }
+ * @param elemProps Props passed into a component
+ * @param componentProps Props the component provides
+ */
+export function mergeCallback<T extends Function>(callback1: T, callback2?: Function): T {
+  if (callback2) {
+    return (((...args: any[]) => {
+      callback2?.apply(null, args);
+      callback1.apply(null, args);
+    }) as any) as T;
+  } else {
+    return callback1;
+  }
+}
```

Comment:
First merge strategy for a callback being defined in the component and also in `elemProps`. Calls both if both exist. Bonus the `@example` shows in intellisense:

<img width="565" alt="Screen Shot 2020-06-04 at 5 32 12 PM" src="https://user-images.githubusercontent.com/338257/83820240-5f6e0e80-a689-11ea-8994-f8ed6d746f7c.png">